src/commands/readme.rs

```diff
@@ -0,0 +1,238 @@
+                )
+                .map_err(|e| format!("{:}", e))?;
+
+            Ok(())
+        }
+        (mode, existing_res) => {
+            let template_path = find_readme_template(&ws.root(), &pkg_path)?;
+            c.shell()
+                .status(
+                    "Generating",
+                    format!(
+                        "Readme for {} (template: {:?})",
+                        &pkg_manifest.name(),
+                        match &template_path {
+                            Some(p) => p.strip_prefix(&root_path).unwrap().to_str().unwrap(),
+                            None => "none found",
+                        }
+                    ),
+                )
+                .map_err(|e| format!("{:}", e))?;
+            let new_readme = &mut generate_readme(&pkg_path, &mut pkg_source, template_path)?;
+            if mode == &GenerateReadmeMode::Append && existing_res.is_ok() {
+                *new_readme = format!("{}\n{}", existing_res.unwrap(), new_readme);
+            }
+            fs::write(readme_path, new_readme.as_bytes()).map_err(|e| format!("{:}", e))
+        }
+    }
+}
+
+fn generate_readme<'a>(
+    pkg_path: &Path,
+    pkg_source: &mut File,
+    template_path: Option<PathBuf>,
+) -> Result<String, String> {
+    let mut template = template_path
+        .map(|p| fs::File::open(&p).expect(&format!("Could not read template at {}", p.display())));
+
+    cargo_readme::generate_readme(
+        pkg_path,
+        pkg_source,
+        template.as_mut(),
+        false,
+        false,
+        true,
+        false,
+    )
+}
+
+/// Find the default entrypoiny to read the doc comments from
+///
+/// Try to read entrypoint in the following order:
+/// - src/lib.rs
+/// - src/main.rs
+/// - file defined in the `[lib]` section of Cargo.toml
+/// - file defined in the `[[bin]]` section of Cargo.toml, if there is only one
+///   - if there is more than one `[[bin]]`, an error is returned
+fn find_entrypoint(current_dir: &Path, manifest: &Manifest) -> Result<File, String> {
+    let entrypoint = find_entrypoint_internal(current_dir, &manifest)?;
+    File::open(current_dir.join(entrypoint)).map_err(|e| format!("{}", e))
+}
+
+/// Find the default entrypoiny to read the doc comments from
+///
+/// Try to read entrypoint in the following order:
+/// - src/lib.rs
+/// - src/main.rs
+/// - file defined in the `[lib]` section of Cargo.toml
+/// - file defined in the `[[bin]]` section of Cargo.toml, if there is only one
+///   - if there is more than one `[[bin]]`, an error is returned
+fn find_entrypoint_internal(current_dir: &Path, _manifest: &Manifest) -> Result<PathBuf, String> {
+    // try lib.rs
+    let lib_rs = current_dir.join("src/lib.rs");
+    if lib_rs.exists() {
+        return Ok(lib_rs);
+    }
+
+    // try main.rs
+    let main_rs = current_dir.join("src/main.rs");
+    if main_rs.exists() {
+        return Ok(main_rs);
+    }
+
+    // try lib defined in `Cargo.toml`
```

Comment:
is this still expected to be here? Either activate or remove maybe?

Reply:
See below.